docs-src/how-to-create-layoutbuilder.md

```diff
@@ -14,10 +14,303 @@ kernelspec:
+
+```{code-cell}
+builder.tag(0)
+builder.float64(1.1)
+builder.tag(1)
+builder.boolean(False)
+builder.tag(2)
+builder.int64(11)
+```
+
+ The contents filling order can be arbitrary. `tag` uniquely identifies the content, the next command fills it.
+
+```{code-cell}
+builder.tag(0)
+builder.float64(2.2)
+builder.tag(1)
+builder.boolean(False)
+builder.tag(0)
+builder.float64(2.2)
+builder.tag(0)
+builder.float64(3.3)
+builder.tag(1)
+builder.boolean(True)
+builder.tag(0)
+builder.float64(4.4)
+builder.tag(1)
+builder.boolean(False)
+builder.tag(1)
+builder.boolean(True)
+builder.tag(0)
+builder.float64(-2.2)
+```
+
+Snapshot
+--------
+
+To turn a `LayoutBuilder` into a layout, call `snapshot`. This is an inexpensive operation (may be done multiple times; the builder is unaffected).
+
+```{code-cell}
+layout = builder.snapshot()
+layout
+```
+
+If you want to use the layout as a high-level array for normal analysis, remember to convert it.
+
+```{code-cell}
+array = ak.Array(layout)
+array
+```
+
+Nested lists
+------------
+
+To fill data inside of a list use the following commands:
+
+  * `begin_list`/`end_list`
+
+Here is an example of a list offset array form:
+
+```{code-cell}
+form = """
+{
+  "class": "ListOffsetArray64",
+  "offsets": "i64",
+  "content": "float64",
+  "form_key": "node0"
+}
+"""
+```
+Create a builder from the form:
+
+```{code-cell}
+builder = ak.layout.LayoutBuilder32(form)
+```
+
+and append the data between `begin_list` and `end_list`:
+
+```{code-cell}
+builder.begin_list()
+builder.float64(1.1)
+builder.float64(2.2)
+builder.float64(3.3)
+builder.end_list()
+```
+
+To append an empty list:
+
+```{code-cell}
+builder.begin_list()
+builder.end_list()
+```
+
+and continue:
+
+```{code-cell}
+builder.begin_list()
+builder.float64(4.4)
+builder.float64(5.5)
+builder.end_list()
+```
+Remember, you can taka a snapshot at any time:
+
+```{code-cell}
+layout = builder.snapshot()
+layout
+```
+
+Nested records
+--------------
+
+When using a RecordArray form you do not have to specify a field, the fields alternate.
+
+```{code-cell}
+form = """
+{
+"class": "RecordArray",
+"contents": {
+    "one": "float64",
+    "two": "int64"
+},
+"form_key": "node0"
+}
+"""
+builder = ak.layout.LayoutBuilder32(form)
+
+# the fields alternate
+builder.float64(1.1)  # "one"
+builder.int64(2)      # "two"
+builder.float64(3.3)  # "one"
+builder.int64(4)      # "two"
+```
+
+```{code-cell}
+layout = builder.snapshot()
+layout
+```
+
+Similarly, for the record contents with the same type:
+
+```{code-cell}
+form = """
+{
+"class": "RecordArray",
+"contents": {
+    "one": "float64",
+    "two": "float64"
+},
+"form_key": "node0"
+}
+"""
+```
+
+```{code-cell}
+builder = ak.layout.LayoutBuilder32(form)
+```
+
+If record contents have the same type, the fields alternate:
+
+```{code-cell}
+builder.float64(1.1)  # "one"
+builder.float64(2.2)  # "two"
+builder.float64(3.3)  # "one"
+builder.float64(4.4)  # "two"
+```
```

Comment:
Can we still specify the field if we want? If so, could we mention that?

````suggestion
```
However, the fields can be explicitly specified if required:
``{code-cell}
builder.field("two").float64(1.1)  # "two"
builder.field("one").float64(2.2)  # "one"
builder.field("one").float64(3.3)  # "one"
builder.field("two").float64(4.4)  # "two"
```
````

Reply:
I don't remember whether `builder.field("xyz")` acts as an assertion or not. It will not allow you to write them in a different order: that's a performance thing.